app/src/main/java/ayds/newyork/songinfo/moredetails/fulllogic/OtherInfoWindow.kt

```diff
@@ -0,0 +1,191 @@
+import androidx.appcompat.app.AppCompatActivity
+import ayds.newyork.songinfo.R
+import ayds.newyork.songinfo.utils.UtilsInjector
+import ayds.newyork.songinfo.utils.view.ImageLoader
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.scalars.ScalarsConverterFactory
+import java.io.IOException
+import java.util.Locale
+
+private const val RESPONSE = "response"
+private const val DOCS = "docs"
+private const val ABSTRACT = "abstract"
+private const val WEB_URL = "web_url"
+private const val PREFIX = "[*]"
+private const val NO_RESULTS = "No Results"
+private const val HTML_START = "<html><div width=400>"
+private const val HTML_FONT = "<font face=\"arial\">"
+private const val HTML_END = "</font></div></html>"
+
+class OtherInfoWindow : AppCompatActivity() {
+
+    private lateinit var moreDetailsTextView: TextView
+    private lateinit var titleImageView: ImageView
+    private lateinit var openUrlButton: Button
+    private lateinit var dataBase: DataBase
+    private val imageLoader: ImageLoader = UtilsInjector.imageLoader
+    private var artistName: String? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_other_info)
+        initProperties()
+        initDataBase()
+        updateTitleImageView()
+        initArtistName()
+        getArtistInfo()
+    }
+
+    private fun initArtistName() {
+        artistName = intent.getStringExtra(ARTIST_NAME_EXTRA)
+    }
+
+    private fun initProperties() {
+        moreDetailsTextView = findViewById(R.id.textMoreDetails)
+        titleImageView = findViewById(R.id.imageView)
+        openUrlButton = findViewById(R.id.openUrlButton)
+    }
+
+    private fun initDataBase() {
+        dataBase = DataBase(this)
+    }
+
+    private fun getArtistInfo() {
+        Thread {
+            val artistInfo = searchArtistInfo()
+            updateArtistInfo(artistInfo)
+        }.start()
+    }
+
+    private fun updateArtistInfo(artistInfo: ArtistInfo?) {
+        if (artistInfo != null) {
+            if (artistInfo.url != null)
+                initListeners(artistInfo.url)
+            if (artistInfo.abstract != null)
+                updateMoreDetailsText(artistInfo.abstract!!)
+        }
+    }
+
+    private fun searchArtistInfo(): ArtistInfo? {
+        var artistInfo = getInfoFromDataBase()
+        when {
+            artistInfo != null -> markArtistInfoAsLocal(artistInfo)
+            else -> {
+                artistInfo = getInfoFromAPI()
+                artistInfo?.let {
+                        dataBase.saveArtistInfo(artistInfo)
+                }
+            }
+        }
+        return artistInfo
+    }
+
+    private fun markArtistInfoAsLocal(artistInfo: ArtistInfo) {
+        artistInfo.isLocallyStored = true
+    }
+
+    private fun getTextFromAbstract(abstract: String?): String {
```

Comment:
Pueden usar una mejora de Kotlin poniendo directamente
```suggestion
    private fun getTextFromAbstract(abstract: String?) = if (abstract != null) {
    } ...
```